source/mesh_deformation/fastscape.cc

```diff
@@ -429,6 +429,11 @@ namespace aspect
           // In later timesteps, we copy h directly from FastScape.
           std::mt19937 random_number_generator(fastscape_seed);
           std::uniform_real_distribution<double> random_distribution(-noise_elevation,noise_elevation);
+          // read sea level from the user defined function or constant value;
+          const double current_sea_level = use_sea_level_function
+              ? sea_level_function.value(Point<1>(time_in_years))
+              : sea_level_constant_value;
```

Comment:
This isn't right. What you want is this:
```suggestion
          // read sea level from the user defined function or constant value;
          sea_level_function.set_time(time_in_years);
          const double current_sea_level = use_sea_level_function
              ? sea_level_function.value(Point<1>()) // use a dummy point -- we only care about the time dependence
              : sea_level_constant_value;
```

Reply:
As this is a const method, I have to remove `sea_level_function.set_time(time_in_years);`. Instead, an update function is added to make this work in the latest commit

Reply:
Oh, I see. That makes sense for the call to `set_time()`. But you should still call `.value()` with a dummy argument. The argument is a spatial point, and has nothing to do with time.

Reply:
Scratch that -- I see that you did that already.